test/e2e/gc_e2e_test.go

```diff
@@ -284,4 +282,91 @@ var _ = Describe("Garbage collection for dependent resources", func() {
 
 	})
 
+	Context("Given a configmap and secret associated with a CSV they should be deleted when the CSV is deleted", func() {
+		const (
+			configmapName = "dependentcfg"
+			secretName    = "dependentsecret"
+		)
+
+		var (
+			owner        v1alpha1.ClusterServiceVersion
+			fetchedOwner *v1alpha1.ClusterServiceVersion
+			propagation  metav1.DeletionPropagation
+			options      metav1.DeleteOptions
+			configMap    *corev1.ConfigMap
+			secret       *corev1.Secret
+		)
+
+		BeforeEach(func() {
+			var err error
+			owner = newCSV("owner", testNamespace, "", semver.MustParse("0.0.0"), nil, nil, newNginxInstallStrategy("test", nil, nil))
+			// create all owners
+			fetchedOwner, err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Create(context.TODO(), &owner, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// wait for csv install to complete
+			Eventually(func() bool {
+				csv, _ := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), fetchedOwner.GetName(), metav1.GetOptions{})
+				return csv.Status.Phase == v1alpha1.CSVPhaseSucceeded
+			}).Should(BeTrue())
```

Comment:
Alternatively, there is a [fetchCSV](https://github.com/operator-framework/operator-lifecycle-manager/blob/d23b7ab12e3925a16bdb961cc9f8b8ddbad01cce/test/e2e/csv_e2e_test.go#L3409) function which waits for a CSV to reach the desired phase.